crates/arrow_sql_gen/src/statement.rs

```diff
@@ -308,33 +317,35 @@ impl InsertBuilder {
                             }
                         }
                     }
-                    _ => unimplemented!(
-                        "Data type mapping not implemented for {}",
-                        column.data_type()
-                    ),
+                    _ => return Result::Err(Error::FailedToCreateInsertStatement { source: format!("Data type mapping not implemented for {}", column.data_type()).into() }) 
                 }
             }
-            insert_stmt.values_panic(row_values);
+            match insert_stmt.values(row_values) {
+                Ok(_) => (),
+                Err(e) => return Result::Err(Error::FailedToCreateInsertStatement { source: Box::new(e) }),
+            
+            }
         }
+        Ok(())
     }
 
     #[must_use]
     pub fn build_postgres(self) -> String {
-        self.build(PostgresQueryBuilder)
+        self.build(PostgresQueryBuilder).unwrap_or_default()
```

Comment:
This should also bubble up the error (here and for `build_sqlite` and `build_mysql`)

Reply:
Fine to create an issue to track and leave this as is